rust/lance-index/src/vector/storage.rs

```diff
@@ -38,6 +38,9 @@ use super::DISTANCE_TYPE_KEY;
 /// </section>
 pub trait DistCalculator {
     fn distance(&self, id: u32) -> f32;
+    fn distance_all(&self) -> Vec<f32> {
+        unimplemented!("Implement this")
```

Comment:
Is there a GH issue for this?

Reply:
It's for IVF_PQ only for now, can implement this for all distance calculator